### PR TITLE
[TRANSLATION] Chinese Traditional (zh-TW) translation update - Oct 2021

### PR DESCRIPTION
**Pull request by eason329**
## Purpose
update Chinese Traditional (zh-TW) translation

## Proposed changes
- Translate strings still have original English text
- Replace and unify terms with one used in Taiwan
- replacing the symbols with full width equivalents when appropriate as stated in PR 3954
- Rename and update for For_Chinese_Translation.txt
- Two separated Chinese translation for `COMCTL32` (which is came from Wine) will be combined into one file 
>ROS supports multiple sublanguages in a single language RC file

-----
please add the 'hacktoberfest-accepted' label to this if this PR fits. Thanks 
> I assume this PR may merge after October

### CODE REVIEW

**Comment by eason329** [comment]
If there is no more review in next 7 days, I will assume this PR can be merged.

**Comment by eason329** [comment]
> If there is no more review in next 7 days, I will ping a maintainer to merge this PR.

7 days has passed, so @binarymaster may you check whether this PR can be merged?

File: dll/shellext/acppage/lang/zh-TW.rc [hunk 1]
```diff
@@ -1,4 +1,10 @@
-/* Traditional Chinese translation by Henry Tang Ih 2016 (henrytang2@hotmail.com) */
+/*
+ * PROJECT:     Application Compatibility Layer Shell Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
```
Comment (binarymaster) [inline dll/shellext/acppage/lang/zh-TW.rc:3]: See https://github.com/reactos/reactos/blob/3b008a87d86eda31ea3177f043dfc4c151668d47/dll/shellext/acppage/ACPPage.cpp
```suggestion
 * PROJECT:     ReactOS Compatibility Layer Shell Extension
 * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
```

File: dll/shellext/devcpux/lang/zh-TW.rc [hunk 2]
```diff
@@ -1,4 +1,9 @@
-/* Translated by Luo Yufan (njlyf2011@hotmail.oom) on 2018-12-24 */
+/*
+ * PROJECT:     ReactOS Processor Shell Extension
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Chinese (Traditional) resource file
+ * TRANSLATOR:  Copyright 2018 Luo Yufan <njlyf2011@hotmail.com>
```
Comment (binarymaster) [inline dll/shellext/devcpux/lang/zh-TW.rc:5]: Didn't added yourself here :)
```suggestion
 * TRANSLATOR:  Copyright 2018 Luo Yufan <njlyf2011@hotmail.com>
 *              Copyright 2021 Chan Chilung <eason066@gmail.com>
```

File: dll/shellext/mydocs/lang/zh-TW.rc [hunk 3]
```diff
@@ -1,8 +1,13 @@
-// Chinese Traditional (zh-TW) resources by Chan Chilung <eason066@gmail.com> 2021
+/*
+ * PROJECT:     ReactOS Mydocs
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
```
Comment (binarymaster) [inline dll/shellext/mydocs/lang/zh-TW.rc:3]: See https://github.com/reactos/reactos/blob/3b008a87d86eda31ea3177f043dfc4c151668d47/dll/shellext/mydocs/mydocs.cpp
```suggestion
 * PROJECT:     ReactOS MyDocs
 * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
```

File: dll/shellext/netshell/lang/zh-TW.rc [hunk 4]
```diff
@@ -1,10 +1,9 @@
 /*
- * PROJECT:    ReactOS Netshell
- * LICENSE:    GPL - See COPYING in the top level directory
- * FILE:       dll/shellext/netshell/lang/zh-TW.rc
- * PURPOSE:    Chinese (Traditional) Language File
- * TRANSLATOR: 2011 Elton Chung aka MfldElton <elton328@gmail.com>
- *             2020 - 2021 Chan Chilung <eason066@gmail.com>
+ * PROJECT:     ReactOS Netshell
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
```
Comment (binarymaster) [inline dll/shellext/netshell/lang/zh-TW.rc:3]: See https://github.com/reactos/reactos/blob/3b008a87d86eda31ea3177f043dfc4c151668d47/dll/shellext/netshell/netshell.cpp
```suggestion
 * PROJECT:     ReactOS Network Shell
 * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
```

File: dll/win32/comctl32/lang/comctl_Tw.rc [hunk 5]
```diff
@@ -1,100 +0,0 @@
...
-  PUSHBUTTON    "說明(&H)",    IDC_HELP_BTN,320,40,70,14
-  PUSHBUTTON    "往上移(&U)",  IDC_MOVEUP_BTN,320,74,70,14
-  PUSHBUTTON    "往下移(&D)",  IDC_MOVEDN_BTN,320,91,70,14
-  LTEXT         "可用按鈕(&V):", -1,4,5,120,10
-  LISTBOX       IDC_AVAILBTN_LBOX,5,17,120,100, LBS_NOTIFY | LBS_OWNERDRAWFIXED | LBS_HASSTRINGS | LBS_NOINTEGRALHEIGHT | LBS_DISABLENOSCROLL | WS_BORDER | WS_VSCROLL | WS_HSCROLL | WS_TABSTOP
-  PUSHBUTTON    "新增(&A) ->", IDOK, 132, 42, 55, 14
-  PUSHBUTTON    "<- 移除(&R)", IDC_REMOVE_BTN,132,62,55,14
-  LTEXT         "工具列按鈕(&T):", -1,194,5,120,10
-  LISTBOX       IDC_TOOLBARBTN_LBOX, 194,17,120,100,LBS_NOTIFY | LBS_OWNERDRAWFIXED | LBS_HASSTRINGS | LBS_NOINTEGRALHEIGHT | LBS_DISABLENOSCROLL | WS_BORDER | WS_VSCROLL | WS_HSCROLL | WS_TABSTOP
-END
```
Comment (binarymaster) [inline dll/win32/comctl32/lang/comctl_Tw.rc:100]: I'm not sure it's correct to delete this file.

File: media/doc/Chinese translation notes.txt [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+H2）請使用港式中文書面語翻譯。只有在特殊情況下才可使用粵文 (Written Cantonese)，並需最大程度保留其可讀性。
+H3）香港繁體中文所使用的字形以《香港小學學習字詞表》為標準。
+H4）所有習慣稱呼（例如: 取消、網絡）需要以香港和／或澳門為標準。如兩地習慣稱呼有異，則以前者為準。
+H5）由於 Wine 不支援香港繁體中文，大部份與 Wine 相關的翻譯須保留英文原文（即不翻譯）。如該檔案有特殊原因而需要翻譯，則需使用正體中文版本的翻譯。翻譯時請跟隨上述規則。
+
+其他翻譯細則可以到 ReactOS Wiki 查看。
+
+祝您翻譯愉快！
+
+（最近更新日期：2021 年 11 月 20 日）
```
Comment (binarymaster) [inline media/doc/Chinese translation notes.txt:56]: If you want to rename this file, better do it in separate commit, for clearer history.

File: base/applications/taskmgr/lang/zh-TW.rc [hunk 7]
```diff
@@ -1,9 +1,9 @@
 /*
- * PROJECT:    ReactOS Task Manager
- * LICENSE:    GPL - See COPYING in the top level directory
- * FILE:       base/applications/taskmgr/lang/zh-TW.rc
- * PURPOSE:    Chinese (Traditional) Language File for taskmgr
- * TRANSLATOR: Elton Chung aka MfldElton <elton328@gmail.com>
+ * PROJECT:     ReactOS Task Manager
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
```
Comment (binarymaster) [inline base/applications/taskmgr/lang/zh-TW.rc:3]: See https://github.com/reactos/reactos/blob/3b008a87d86eda31ea3177f043dfc4c151668d47/base/applications/taskmgr/taskmgr.c
```suggestion
 * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
```

File: base/applications/rapps/lang/zh-TW.rc [hunk 8]
```diff
@@ -1,5 +1,10 @@
-/* Translated by Song Fuchang (0xfc) <sfc_0@yahoo.com.cn>
- * Translation improved by Chan Chilung <eason066@gmail.com> */
+/*
+ * PROJECT:     ReactOS ReactOS Applications Manager
```
Comment (binarymaster) [inline base/applications/rapps/lang/zh-TW.rc:2]:
```suggestion
 * PROJECT:     ReactOS Applications Manager
```

File: base/applications/games/spider/lang/zh-TW.rc [hunk 9]
```diff
@@ -1,9 +1,9 @@
 /*
- * PROJECT:    Spider Solitaire
- * LICENSE:    See COPYING in top level directory
- * FILE:       base/applications/games/spider/lang/zh-TW.rc
- * PURPOSE:    Chinese (Traditional) Language File for Spider Solitaire
- * TRANSLATOR: Elton Chung aka MfldElton <elton328@gmail.com>
+ * PROJECT:     ReactOS Spider Solitaire
```
Comment (binarymaster) [inline base/applications/games/spider/lang/zh-TW.rc:2]: Leave as is
```suggestion
 * PROJECT:     Spider Solitaire
```

File: base/applications/dxdiag/lang/zh-TW.rc [hunk 10]
```diff
@@ -1,10 +1,9 @@
 /*
- * PROJECT:    ReactOS ReactX-Diagnostic program
- * LICENSE:    GPL - See COPYING in the top level directory
- * FILE:       base/applications/dxdiag/lang/zh-TW.rc
- * PURPOSE:    Chinese (Traditional) Language File for ReactX-Diagnostic program
- * TRANSLATOR: Elton Chung <elton328@gmail.com>
- *             Chan Chilung <eason066@gmail.com>
+ * PROJECT:     ReactOS ReactX-Diagnostic program
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
```
Comment (binarymaster) [inline base/applications/dxdiag/lang/zh-TW.rc:3]: Actually LGPL - https://github.com/reactos/reactos/blob/3b008a87d86eda31ea3177f043dfc4c151668d47/base/applications/dxdiag/dxdiag.c
```suggestion
 * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
```

File: base/applications/dxdiag/lang/zh-TW.rc [hunk 11]
```diff
@@ -1,10 +1,9 @@
 /*
- * PROJECT:    ReactOS ReactX-Diagnostic program
- * LICENSE:    GPL - See COPYING in the top level directory
- * FILE:       base/applications/dxdiag/lang/zh-TW.rc
- * PURPOSE:    Chinese (Traditional) Language File for ReactX-Diagnostic program
- * TRANSLATOR: Elton Chung <elton328@gmail.com>
- *             Chan Chilung <eason066@gmail.com>
+ * PROJECT:     ReactOS ReactX-Diagnostic program
```
Comment (binarymaster) [inline base/applications/dxdiag/lang/zh-TW.rc:2]:
```suggestion
 * PROJECT:     ReactX Diagnosis Application
```

**Review by binarymaster — CHANGES_REQUESTED** [review]
:warning: I cannot apply suggestions to your PR. Please allow edits from maintainers.

```
 ! [remote rejected]         pr_branch -> patch-15 (permission denied)
error: failed to push some refs to 'https://github.com/eason329/reactos'
```

**Comment by binarymaster** [comment]
Ok I'm going to do it with a temporary branch strategy.

File: dll/win32/comctl32/lang/comctl_Zh.rc [hunk 12]
```diff
@@ -98,3 +98,83 @@ BEGIN
   LTEXT         "当前工具栏按钮(&T):", -1,194,5,120,10
   LISTBOX       IDC_TOOLBARBTN_LBOX, 194,17,120,100,LBS_NOTIFY | LBS_OWNERDRAWFIXED | LBS_HASSTRINGS | LBS_NOINTEGRALHEIGHT | LBS_DISABLENOSCROLL | WS_BORDER | WS_VSCROLL | WS_HSCROLL | WS_TABSTOP
 END
+
+LANGUAGE LANG_CHINESE, SUBLANG_CHINESE_TRADITIONAL
```
Comment (binarymaster) [inline dll/win32/comctl32/lang/comctl_Zh.rc:102]: Ok I verified with Wine, even them have two separate files for zh-CN and zh-TW accordingly:

- https://source.winehq.org/git/wine.git/blob/f69d4a865f926aa5c4a9c55cfe4d2dbc10746e5c:/po/zh_CN.po#l408
- https://source.winehq.org/git/wine.git/blob/f69d4a865f926aa5c4a9c55cfe4d2dbc10746e5c:/po/zh_TW.po#l403

**Comment by binarymaster** [comment]
Finally merged into c1c59624 and 21d71c28, thanks!

**Comment by eason329** [comment]
> ⚠️ I cannot apply suggestions to your PR. Please allow edits from maintainers.
> 
> ```
>  ! [remote rejected]         pr_branch -> patch-15 (permission denied)
> error: failed to push some refs to 'https://github.com/eason329/reactos'
> ```

I can't respond at that moment because that's actually midnight for me and I have exam in the following day. sorry for that

> Ok I verified with Wine, even them have two separate files for zh-CN and zh-TW accordingly:
> 
> * https://source.winehq.org/git/wine.git/blob/f69d4a865f926aa5c4a9c55cfe4d2dbc10746e5c:/po/zh_CN.po#l408
> * https://source.winehq.org/git/wine.git/blob/f69d4a865f926aa5c4a9c55cfe4d2dbc10746e5c:/po/zh_TW.po#l403

I am actually thinking to separating zh translation file for Wine into two, but the veto on [CORE-17110](https://jira.reactos.org/browse/CORE-17110) may prevent me from doing this. I will discuss this on my next PR / on the chat.

**Pull request merged**